src/components/DiagramFrame/DiagramFrame.tsx

```diff
@@ -0,0 +1,267 @@
+  const [showViewOptions, setShowViewOptions] = React.useState(false)
+  const [showNoAside, setShowNoAside] = React.useState(false)
+  const [reload, setReload] = React.useState(false)
+  const [selectionInfo, setSelectionInfo] = React.useState<SelectionInfoPacket>({})
+  const [zoomFactor, setZoomFactor] = React.useState(ZOOM_INIT)
+  const [viewPosition, setViewPosition] = React.useState({x: X_INIT, y: Y_INIT})
+  const [minimapEnabled, setMinimapEnabled] = React.useState(false)
+  const [minimapUntoggled, setMinimapUntoggled] = React.useState(true)
+  const currentModel = useCurrentModel(selectedBranch, modelError)
+
+  const handleHiddenToggle = (event: any) => setHiddenToggle(event.target.checked)
+
+  function showDiagram() {
+    setShowHelp(false)
+    setShowViewOptions(false)
+    setShowSettings(false)
+    setShowNoAside(true)
+    return true
+  }
+
+  function toggleExploreInfo() {
+    if (currentExplore && modelDetail) {
+      Object.keys(selectionInfo).length === 0 || selectionInfo.lookmlElement !== "explore"
+      ? setSelectionInfo({
+        lookmlElement: "explore"
+      })
+      : setSelectionInfo({})
+    }
+  }
+
+  let modelDetails = unfilteredModels ? unfilteredModels.filter((d: ILookmlModel)=>{ 
+    return d.explores.length >= 1
+  }).map((d: ILookmlModel)=>{
+    return {
+      value: d.name,
+      label: d.label
+    }
+  }).sort((a: SelectOptionProps, b: SelectOptionProps) => a.label < b.label ? -1 : 1) : []
+
+  let currentExplore: ILookmlModelExplore = modelDetail && modelDetail.explores.filter((d: ILookmlModelExplore)=>{
+    return d.name === pathExploreName
+  })[0]
+
+  let exploreList: ExploreDropdown[] = currentModel && currentModel.explores.map((d: ILookmlModelExplore)=>{
+    return {
+      value: d.name,
+      label: d.label
+    }
+  }).sort((a: ExploreDropdown, b: ExploreDropdown) => a.label < b.label ? -1 : 1)
+
+  let currentDimensions: DiagrammedModel = dimensions && dimensions.filter((d: DiagrammedModel)=>{
+    return d.exploreName === pathExploreName
+  })[0]
+
+  let currentDiagramMetadata: DiagramMetadata = currentDimensions && currentDimensions.diagramDict
+
+  let defaultViews: VisibleViewLookup = {}
+  Object.keys(viewVisible).length === 0 && currentExplore && currentDiagramMetadata && Object.keys(currentDiagramMetadata.tableData)
+  .map((lookmlViewName: string)=>{
+    defaultViews[lookmlViewName] = true
+  })
+  Object.keys(viewVisible).length === 0 && currentExplore && currentDiagramMetadata && setViewVisible(defaultViews)
+
+  return (
+    <Layout hasAside height="100%">
+      <Rail width="50px">
+        <SpaceVertical style={{alignItems: "center"}} alignItems="center" gap="xsmall">
+          <IconButton
+            icon="Dashboard"
+            label="Diagram"
+            tooltipPlacement="right"
+            size="large"
+            toggle={showNoAside && !showViewOptions && !showSettings && !showHelp ? true : undefined}
```

Comment:
Another general point I won't comment on further, this kind of thing is much better in named functions. It's hard to understand a series of conditions like this, while a function could specifically tell me under what condition this toggle should occur. Even if it is nebulous like this, you can docstring it for the next person.